.github/workflows/CI.yml

```diff
@@ -92,17 +92,39 @@ jobs:
         with:
           java-version: ${{ matrix.java }}
 
-      - name: Install dependencies on macos
+      - name: Build native libraries
         run: |
+          git submodule update --init --recursive
           brew reinstall gcc
-          export FC=/usr/local/Cellar/gcc/12.2.0/bin/gfortran
+          brew install llvm
+          brew install openblas
+          export FC=/opt/homebrew/bin/gfortran
+          export PATH=/opt/homebrew/opt/llvm/bin:$PATH
+          export CC=/opt/homebrew/opt/llvm/bin/clang
+          export CXX=/opt/homebrew/opt/llvm/bin/clang++
+          cd jni
+          sed -i -e 's/\/usr\/local\/opt\/libomp\//\/opt\/homebrew\/opt\/llvm\//g' cmake/init-faiss.cmake
+          sed -i -e 's/__aarch64__/__undefine_aarch64__/g' external/faiss/faiss/utils/distances_simd.cpp
```

Comment:
This should be possible to avoid. In faiss, I think they are building it without patches: https://github.com/facebookresearch/faiss/pull/3411.